### ISSUE TITLE
If vuex action triggers interceptor the returned value is undefined instead of the actual interceptor response

### ISSUE THREAD

**Issue by wooooodward** [nuxt-community/axios-module#238]
### Version

[5.4.1](https://github.com/axios-module/releases/tag/5.4.1)

### Reproduction link

[https://codesandbox.io/s/moywxno29p](https://codesandbox.io/s/moywxno29p)

### Steps to reproduce

Load the codesandbox link then click Go to Posts. On page load here's what happens:

1. Call vuex action using the `fetch` method
2. Action triggers 400 level error
3. Intercept error using `onResponseError` in plugin
4. Retry the request after performing some other actions

### What is expected ?

The interceptor successfully returns some data and the action continues, committing that data to the store.

### What is actually happening?

Despite the interceptor having a successful retry and returning valid data, the axios call in the store action returns undefined.

In other words `await.$axios.$post(...)` triggers an error which is successfully intercepted but is `undefined`.

It should be equal to the value returned by the interceptor.

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/axios-module/issues/c231">#c231</a>)</em></sub></div>

**Comment by wooooodward** [comment]
Wow, figured it out shortly after posting this.

The axios module's methods prefixed with `$` automatically return `response.data`

Instead of returning in the interceptor with

    return $axios.$request({
        method: error.response.config.method,
        url: error.response.config.url,
        data: error.response.config.data
    })

You need to return a dict with data as a key

    return {
        data: $axios.$request({
            method: error.response.config.method,
            url: error.response.config.url,
            data: error.response.config.data
        })
    }

**Comment by ricardogobbosouza** [comment]
They are helpers for the axios module
https://axios.nuxtjs.org/helpers#fetch-style-requests
